src/tensorlake/function_executor/service.py

```diff
@@ -41,13 +44,87 @@
     InitializeResponse,
     InvocationStateRequest,
     InvocationStateResponse,
+    ListTasksRequest,
+    ListTasksResponse,
+    ProgressUpdate,
     RunTaskRequest,
     RunTaskResponse,
+    Task,
+    TaskFailureReason,
+    TaskOutcomeCode,
+    TaskResult,
 )
 from .proto.function_executor_pb2_grpc import FunctionExecutorServicer
 from .std_outputs_capture import flush_logs, read_till_the_end
 
 
+def _run_task_request_to_function_inputs(run_request: RunTaskRequest) -> FunctionInputs:
+    """Convert RunTaskRequest to FunctionInputs."""
+    request = FunctionInputs()
+
+    if run_request.HasField("function_input"):
+        request.function_input.CopyFrom(run_request.function_input)
```

Comment:
This is a required field, there's helper MessageValidator class that can do this validation.

Reply:
This code will be removed pretty soon, and this structure lets us do validation in one place.